src/main/scala/edu/berkeley/ce/rockslicing/LoadBalancer.scala

```diff
@@ -0,0 +1,323 @@
+    *                      Should be one less than number of cores used.
+    * @return Set of processor joints that will be used to seed the initial RDD
+    */
+  def generateSeedJoints(rockVolume: Block, numSeedJoints: Integer):
+                         Seq[Joint] = {
+    // Calculate bounding box
+    val vertices = rockVolume.findVertices.values.flatten
+    val x_max = (vertices.map{ case vertex => vertex._1}).max
+    val x_min = (vertices.map{ case vertex => vertex._1}).min
+    val y_max = (vertices.map{ case vertex => vertex._2}).max
+    val y_min = (vertices.map{ case vertex => vertex._2}).min
+    val z_max = (vertices.map{ case vertex => vertex._3}).max
+    val z_min = (vertices.map{ case vertex => vertex._3}).min
+    val boundingBox = (x_min, y_min, z_min, x_max, y_max, z_max)
+
+    // Diagonal vector from lower left to upper right corner of bounding box
+    val diagonalVector = breeze.linalg.normalize(DenseVector[Double](boundingBox._4 - boundingBox._1,
+                                                                     boundingBox._5 - boundingBox._2,
+                                                                     boundingBox._6 - boundingBox._3))
+    val volumePerPart = rockVolume.volume/(numSeedJoints + 1)
+    val centroid = rockVolume.centroid
+    val centroidVolume = Block(centroid, rockVolume.updateFaces(centroid))
+    val seedJoints =
+      findProcessorJoints(Seq[Joint](), (centroidVolume.centerX, centroidVolume.centerY, centroidVolume.centerZ),
+                          (boundingBox._1, boundingBox._2, boundingBox._3),
+                          centroidVolume, volumePerPart)
+
+    if (seedJoints.length != numSeedJoints) {
+      throw new IllegalStateException("ERROR: LoadBalancer.generateSeedJoints unable to find optimal "+
+                                      "seed joints")
+    }
+    seedJoints
+  }
+
+  /**
+    * Finds the best set of processor joints to divide initial volume into approximately equal volumes
+    * @param joints List to prepend processor joints to during each recursion
+    * @param origin Origin which the joint is in reference to
+    * @param center Center of the joint plane - start with lower left corner of bounding box
+    * @param initialVolume Intial volume that needs to be sub-divided
+    * @param desiredVolume Desired volume for each piece cut by processor joints
+    * @return List of processor joints that will be used to seed initial RDD
+    */
+  @tailrec
+  private def findProcessorJoints(joints: Seq[Joint],
+                                  origin: (Double, Double, Double),
+                                  center: (Double, Double, Double), initialVolume: Block,
+                                  desiredVolume: Double):
+                                  Seq[Joint] = {
+    // Calculate bounding box
+    val vertices = initialVolume.findVertices.values.flatten
+    val x_max = (vertices.map{ case vertex => vertex._1}).max
```

Comment:
Like before, I think we can do `map(_._1)` and similar for the other extrema instead of using `case`.